src/Microsoft.Diagnostics.Runtime/src/DataTargets/DataTarget.cs

```diff
@@ -237,88 +235,102 @@ public IEnumerable<ModuleInfo> EnumerateModules()
             RuntimeInformation.IsOSPlatform(OSPlatform.Linux) ? new LinuxFunctions() :
             RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ? new MacOSFunctions() :
             (PlatformFunctions)new WindowsFunctions();
-
+        
         /// <summary>
-        /// Loads a dump file.  Currently supported formats are ELF coredump and Windows Minidump formats.
+        /// Loads a dump stream. Currently supported formats are ELF coredump and Windows Minidump formats.
         /// </summary>
-        /// <param name="filePath">The path to the dump file.</param>
-        /// <param name="cacheOptions">The caching options to use </param>
-        /// <returns>A <see cref="DataTarget"/> for the given dump file.</returns>
-        public static DataTarget LoadDump(string filePath, CacheOptions? cacheOptions = null)
+        /// <param name="displayName">The name of this DataTarget, might be used in exceptions.</param>
+        /// <param name="stream">The stream that should be used. Ownership will be transferred to the DataTarget.</param>
+        /// <param name="cacheOptions">The caching options to use. (Only used for FileStreams)</param>
+        /// <returns>A <see cref="DataTarget"/> for the given dump.</returns>
+        public static DataTarget LoadDumpFromStream(string displayName, Stream stream, CacheOptions? cacheOptions = null)
```

Comment:
If we provide a `Stream` constructor, we should also take a `leaveOpen` flag similar to other C# apis.  That would have to be plumbed through all of the data readers to make sure that they respect whether to leave the stream open or dispose of it.